tuf/api/metadata.py

```diff
@@ -1153,17 +1154,19 @@ def from_dict(cls, delegations_dict: Dict[str, Any]) -> "Delegations":
         for keyid, key_dict in keys.items():
             keys_res[keyid] = Key.from_dict(keyid, key_dict)
         roles = delegations_dict.pop("roles")
-        roles_res = []
+        roles_res: "OrderedDict[str, DelegatedRole]" = OrderedDict()
         for role_dict in roles:
             new_role = DelegatedRole.from_dict(role_dict)
-            roles_res.append(new_role)
+            if roles_res.get(new_role.name) is not None:
```

Comment:
this works (and is probably used elsewhere) but I guess the pythonic way is
`if new_role.name in roles_res:`